### DISCUSSION TITLE
Scan host list to see which one detects sni (netfree)

### DISCUSSION THREAD

**Discussion by demianrey** [projectdiscovery/httpx#741] [Q&A]
Hi, Well, that there is an option where in a list of Hosts it shows which one supports sni, this is used for netfree in mobile operators, here is an example of the code that can help you, credits to the creator
https://github.com/aztecrabbit/bugscanner-go/blob/master/cmd/scan_sni.go

**Comment by ehsandeep** [comment]
@demianrey could you elaborate more about this feature? what needs to be achieved and its use cases as its unclear based on shared information.

**Comment by demianrey** [comment]
> @demianrey could you elaborate more about this feature? what needs to be achieved and its use cases as its unclear based on shared information.

Ok I'll try to explain, this is used to find Hosts that give sni connection in netfree, so it should be used when you don't have data in your cellular company, as Host of social networks if the plan or package has unlimited social networks or Host of the company's portals, etc.
So what is needed? A list of possible Hosts, and that from that list only show those that give sni connection... I don't know if that's enough or if you want something more technical, the technical stuff is in the file I sent.
Regards

**Comment by demianrey** [comment]
Example:
![IMG_20220702_112851](https://user-images.githubusercontent.com/48655207/177008726-37e0dcb9-947b-4f16-9725-5fc44cd96b00.jpg)

**Comment by Mzack9999** [comment]
@demianrey Thanks for opening this issue. `httpx` already support the SNI tls extension automatically by using the `Host` header name if available or a custom value provided by the user, but it's specific for HTTP protocol. Unless I'm mistaken, `tlsx` might be what you need with the SNI bruteforce functionality implemented at https://github.com/projectdiscovery/tlsx/pull/46

**Comment by demianrey** [comment]
> @demianrey Thanks for opening this issue. `httpx` already support the SNI tls extension automatically by using the `Host` header name if available or a custom value provided by the user, but it's specific for HTTP protocol. Unless I'm mistaken, `tlsx` might be what you need with the SNI bruteforce functionality implemented at [projectdiscovery/tlsx#46](https://github.com/projectdiscovery/tlsx/pull/46)

Thanks for your answer friend, I tried, but I couldn't find the command that does what I need... could you tell me how it is, please

**Comment by Mzack9999** [comment]
- **httpx**: `echo x.x.x.x | httpx -sni hostname`
- **tlsx**: `echo google.com | ./tlsx -vc -sni sni_list.txt -tps -silent`
